core/domain/stats_services.py

```diff
@@ -1024,3 +1026,239 @@ def _get_calc_output(exploration_id, state_name, calculation_id):
             calculation_output)
     else:
         return None
+
+
+def get_learner_answer_details_from_model(learner_answer_details_model):
+    """Returns a LearnerAnswerDetails domain object given a
+    LearnerAnswerDetailsModel loaded from the datastore.
+
+    Args:
+        learner_answer_details_model: LearnerAnswerDetailsModel. The learner
+            answer details model loaded from the datastore.
+
+    Returns:
+        LearnerAnswerDetails. A LearnerAnswerDetails domain object
+            corresponding to the given model.
+    """
+    return stats_domain.LearnerAnswerDetails(
+        learner_answer_details_model.state_reference,
+        learner_answer_details_model.entity_type,
+        learner_answer_details_model.interaction_id,
+        [stats_domain.LearnerAnswerInfo.from_dict(learner_answer_info_dict)
+         for learner_answer_info_dict
+         in learner_answer_details_model.learner_answer_info_list],
+        learner_answer_details_model.learner_answer_info_schema_version,
+        learner_answer_details_model.accumulated_answer_info_json_size_bytes)
+
+
+def get_learner_answer_details(state_reference, entity_type):
+    """Returns a LearnerAnswerDetails domain object, with given
+    state_reference, entity_type and interaction_id. This function checks
+    in the datastore if the corresponding LearnerAnswerDetailsModel exists,
+    if not then None is returned.
+
+    Args:
+        state_reference: str. This is used to refer to a state
+            in an exploration or question. For an exploration the
+            value will be equal to 'exp_id.state_name' and for question
+            this will be equal to 'question_id'.
+        entity_type: str. The type of entity i.e ENTITY_TYPE_EXPLORATION
+            or ENTITY_TYPE_QUESTION, which are declared in feconf.py.
+
+    Returns:
+        LearnerAnswerDetails or None. The learner answer domain object
+            or None if the model does not exist.
+    """
+    learner_answer_details_model = (
+        stats_models.LearnerAnswerDetailsModel.get_model_instance(
+            entity_type, state_reference))
+    if learner_answer_details_model:
+        learner_answer_details = get_learner_answer_details_from_model(
+            learner_answer_details_model)
+        return learner_answer_details
+    else:
```

Comment:
This else isn't needed, if the above condition is executed, because of the return, it wouldn't come here anyway.

Reply:
Done